components/renku_data_services/session/blueprints.py

```diff
@@ -0,0 +1,92 @@
+"""Session blueprint."""
+
+from dataclasses import dataclass
+from datetime import datetime, timezone
+
+from sanic import HTTPResponse, Request, json
+from sanic_ext import validate
+
+import renku_data_services.base_models as base_models
+from renku_data_services.base_api.auth import authenticate, only_authenticated
+from renku_data_services.base_api.blueprint import BlueprintFactoryResponse, CustomBlueprint
+from renku_data_services.errors import errors
+from renku_data_services.session import apispec, models
+from renku_data_services.session.db import SessionRepository
+from renku_data_services.users.db import UserRepo
+
+
+@dataclass(kw_only=True)
+class SessionsBP(CustomBlueprint):
+    """Handlers for manipulating sessions."""
+
+    session_repo: SessionRepository
+    user_repo: UserRepo
+    authenticator: base_models.Authenticator
+
+    def get_all(self) -> BlueprintFactoryResponse:
+        """List all sessions."""
+
+        @authenticate(self.authenticator)
+        async def _get_all(_: Request, *, user: base_models.APIUser):
+            sessions = await self.session_repo.get_sessions(user=user)
+            return json(
+                [apispec.Session.model_validate(p).model_dump(exclude_none=True, mode="json") for p in sessions], 201
```

Comment:
This should send back a 200:

```suggestion
                [apispec.Session.model_validate(p).model_dump(exclude_none=True, mode="json") for p in sessions]
```